ansible_collections/arista/avd/plugins/plugin_utils/schema/avddataconverter.py

```diff
@@ -0,0 +1,230 @@
+                schema_path=key,
+            )
+
+
+def _items(validator, items, instance, schema):
+    '''
+    This function performs conversion on each item with the items subschema
+    '''
+    if not validator.is_type(instance, "array"):
+        return
+
+    # Don't run validator if $ref is part of the schema.
+    # Instead $ref validator will pop $ref and run all validators.
+    if "$ref" in schema:
+        return
+
+    # Resolve $ref for items, to support schema actions below which operates on the child schema
+    if "$ref" in items:
+        scope, resolved = validator.resolver.resolve(items["$ref"])
+        merged_childschema = copy.deepcopy(resolved)
+        always_merger.merge(merged_childschema, items)
+        merged_childschema.pop("$ref", None)
+        items = merged_childschema
+
+    # Perform type conversion of instance items if required based on "convert_types"
+    if "convert_types" in items:
+        for index, instance_item in enumerate(instance):
+            instance[index] = _convert_types(validator, items["convert_types"], instance_item, items)
+
+    # Perform regular validation of child elements.
+    # Using schema ["items"] to use the original schema instead of the $ref merged one.
+    for index, instance_item in enumerate(instance):
+        yield from validator.descend(
+            instance=instance_item,
+            schema=schema["items"],
+            path=index,
+        )
+
+
+def _ref(validator, ref, instance, schema: dict):
+    '''
+    This function resolves the $ref referenced schema,
+    then merges with any schema defined at the same level
+    Then performs validation on the resolved+merged schema.
+
+    Since this will run all validation tasks on the same level,
+    a check for $ref has been added to the other validators, to
+    avoid duplicate validation (and duplicate errors)
+    '''
+    scope, resolved = validator.resolver.resolve(ref)
+    validator.resolver.push_scope(scope)
+    merged_schema = copy.deepcopy(resolved)
+    always_merger.merge(merged_schema, schema)
+    merged_schema.pop("$ref", None)
+    try:
+        yield from validator.descend(instance, merged_schema)
+    finally:
+        validator.resolver.pop_scope()
+
+
+def _convert_types(validator, convert_types, instance, schema):
+    '''
+    This function performs type convertion if necessary on a single data instance.
+    It is invoked for child keys during "keys" conversion and for child items during
+    "items" conversion
+    Returns the converted value is returned to the calling converter.
+    Any convertion errors are ignored and the original value is returned
+    '''
+    schema_type = schema.get('type')
+    # Skip conversion if the value is of the correct type already
+    if validator.is_type(instance, schema_type):
+        return instance
+
+    simple_converters = {
+        "str": str,
+        "int": int,
+        "bool": bool,
+    }
+
+    for convert_type in convert_types:
```

Comment:
It would not convert, but since conversion happens before validation, we would catch the type error in the validation. I didn't want the conversion to be the one triggering errors left and right. Comments?

Reply:
I will add messages for both successful and failed conversions